Packs/EmailCommunication/Scripts/SendEmailReply/SendEmailReply_test.py

```diff
@@ -144,3 +159,105 @@ def test_get_email_cc(current_cc, additional_cc, excepted):
     from SendEmailReply import get_email_cc
     result = get_email_cc(current_cc, additional_cc)
     assert result == excepted
+
+
+def test_create_thread_context(mocker):
+    email_threads = util_load_json('test_data/email_threads.json')
+    # demisto.executeCommand will be called twice in the tested function - prepare separate responses for each
+
+    def side_effect_function(command, args):
+        if command == "getContext":
+            return email_threads
+        elif command == "executeCommandAt":
+            return True
+
+    from SendEmailReply import create_thread_context
+
+    # Email data to use both for function input and function output validation
+    test_email = email_threads[0]
+    # Mock function to get current time string to match the expected result
+    mocker.patch('SendEmailReply.get_utc_now',
+                 return_value=datetime.strptime(test_email['MessageTime'], "%Y-%m-%dT%H:%M:%SUTC"))
+
+    execute_command_mocker = mocker.patch.object(demisto, 'executeCommand', side_effect=side_effect_function)
+    create_thread_context(test_email['EmailCommsThreadId'], test_email['EmailCC'], test_email['EmailBCC'],
+                          test_email['EmailBody'], test_email['EmailFrom'], test_email['EmailHTML'],
+                          test_email['MessageID'], test_email['EmailReceived'], test_email['EmailReplyTo'],
+                          test_email['EmailSubject'], test_email['EmailTo'], '123', ['None'])
+    call_args = execute_command_mocker.call_args
+    assert test_email == call_args.args[1]['arguments']['value']
+
+
+@pytest.mark.parametrize(
+    "new_thread, thread_input_type",
+    [('n/a', ''),
+     ('true', ''),
+     ('false', 'dict'),
+     ('false', 'list')]
+)
+def test_main(new_thread, thread_input_type, mocker):
+    """Unit test
+    Given
+     - new_thread = 'n/a'
+     - new_thread = 'true'
+     - new_thread = 'false'
+    When
+        Preparing email message to send
+    Then
+     - validate that logic is followed as expected
+     - for case with new_thread = 'false', validate proper email details are pulled from context
```

Comment:
I've added a separate block of information for each permutation of the test function.  This makes the documentation block for this one function quite long, but there's enough re-used setup for this test that it makes sense to have a single function than have a separate one for each scenario.

>  Unit test scenario - new_thread = 'n/a'
>         Given
>          - Script is called to send an email message
>         When
>          - new_thread = 'n/a', indicating use by the generic 'Email Communication' layout workflow replying to message
>         Then
>          - validate that the new email is constructed and the send_reply function is called successfully
>     Unit test scenario - new_thread = 'true'
>         Given
>          - Script is called to send an email message
>         When
>          - new_thread = 'true', indicating use by incident types besides 'Email Communication' and sending a
>          first-contact outbound email message
>         Then
>          - validate that the new email is constructed and the send_new_email function is called successfully
>     Unit test scenario - new_thread = 'false'
>         Given
>          - Script is called to send an email message
>         When
>          - new_thread = 'false', indicating use by incident types besides 'Email Communication' and sending a
>          reply to an existing email chain
>          - 'EmailThreads' context key is type dict, indicating an attempt to reply to an email chain that has never
>          received a reply from the end user and we must send a new email message rather than reply to one
>         Then
>          - validate that the new email is constructed and resend_first_contact is called successfully
>     Unit test scenario - new_thread = 'false'
>         Given
>          - Script is called to send an email message
>         When
>          - new_thread = 'false', indicating use by incident types besides 'Email Communication' and sending a
>          reply to an existing email chain
>          - 'EmailThreads' context key is type list, indicating there is at least one reply to the first-contact email
>         Then
>          - validate that the new email is constructed, re-using correct items from the existing email thread entries
>          - validate that send_reply is called successfully
>          - validate that create_thread_context is successfully called